packages/node/src/BacktraceNodeRequestHandler.ts

```diff
@@ -0,0 +1,155 @@
+                        rejectUnauthorized: this._options.ignoreSslCertificate === true,
+                        timeout: this._timeout,
+                        method: 'POST',
+                        headers: typeof payload === 'string' ? this.JSON_HEADERS : payload.getHeaders(),
+                    },
+                    (response) => {
+                        let result = '';
+                        response.on('data', (d) => {
+                            result += d.toString();
+                        });
+                        response.on('end', () => {
+                            switch (response.statusCode) {
+                                case 200: {
+                                    res(BacktraceReportSubmissionResult.Ok(this.parseServerResponse(result)));
+                                    break;
+                                }
+                                case 401:
+                                case 403: {
+                                    res(BacktraceReportSubmissionResult.OnInvalidToken());
+                                    break;
+                                }
+                                case 429: {
+                                    res(BacktraceReportSubmissionResult.OnLimitReached());
+                                    break;
+                                }
+                                default: {
+                                    res(BacktraceReportSubmissionResult.OnInternalServerError(result));
+                                    break;
+                                }
+                            }
+                        });
+                    },
+                );
+
+                request.on('error', (err: Error) => {
+                    if (ConnectionError.isConnectionError(err)) {
+                        return res(BacktraceReportSubmissionResult.OnNetworkingError(err.message));
+                    }
+                    return res(BacktraceReportSubmissionResult.OnInternalServerError(err.message));
+                });
+
+                if (typeof payload === 'string') {
+                    request.write(payload);
+                } else {
+                    payload.pipe(request);
+                }
+
+                request.end();
+            });
+        } catch (err) {
+            if (ConnectionError.isConnectionError(err)) {
+                return BacktraceReportSubmissionResult.OnNetworkingError(err.message);
+            }
+
+            const errorMessage = err instanceof Error ? err.message : (err as string);
+            return BacktraceReportSubmissionResult.OnUnknownError(errorMessage);
+        }
+    }
+
+    private getHttpClient(submissionUrl: URL) {
+        return submissionUrl.protocol === 'http' ? http : https;
+    }
+
+    private parseServerResponse<T>(response: string): T {
+        try {
+            return JSON.parse(response);
+        } catch {
+            return {} as T;
+        }
```

Comment:
Should we catch JSON failures? Maybe let the request catch handle this?

Reply:
I had mixed feelings about that because the report is there and with callback apis, there is always a chance we won't be fast enough. This is why I wanted to still anyway tell user everything went ok, but the output might be unexpected. 


I changed the implementation and allow now parsing errors. We will revisit it once we recognize this is the issue for our users